__tests__/app.test.js

```diff
@@ -167,5 +208,88 @@ describe('Error-handling Sad paths', () => {
                     })
                 })
         });
+        it('POST /api/articles/:article_id/comments should return a status 404 and a "not found" message if a valid format, non-existent ID is provided', () => {
+            const newComment = {
+                username: "rogersop",
+                body: "I am posting this comment to this api to see if it works."
+            }
+            return request(app)
+                .post("/api/articles/5823/comments")
+                .send(newComment)
+                .expect(404)
+                .then(({ body }) => {
+                    expect(body).toStrictEqual({
+                        message: "Not Found",
+                        status: 404
+                    })
+                })
+        });
+    });
+    describe('Incorrect format object sent', () => {
+        describe('POST /api/articles/:article_id/comments should return a status 400 and a "Bad Request" message if an incorrectly formatted object is sent' , () => {
+            it('case 1: invalid keys provided', () => {
+                const newComment = {
+                    user: "rogersop",
+                    body: "I am posting this comment to this api to see if it works."
+                }
+                return request(app)
+                    .post("/api/articles/1/comments")
+                    .send(newComment)
+                    .expect(400)
+                    .then(({ body }) => {
+                        expect(body).toStrictEqual({
+                            message: "Bad Request",
+                            status: 400
+                        })
+                    })
+            });
+            it('case 2: values provided in invalid format', () => {
+                const newComment = {
+                    username: "rogersop",
+                    body: 54
+                }
+                return request(app)
+                    .post("/api/articles/1/comments")
+                    .send(newComment)
+                    .expect(400)
+                    .then(({ body }) => {
+                        expect(body).toStrictEqual({
+                            message: "Bad Request",
+                            status: 400
+                        })
+                    })
+            });
+            it('case 3: missing key in object', () => {
+                const newComment = {
+                    username: "rogersop"
+                }
+                return request(app)
+                    .post("/api/articles/1/comments")
+                    .send(newComment)
+                    .expect(400)
+                    .then(({ body }) => {
+                        expect(body).toStrictEqual({
+                            message: "Bad Request",
+                            status: 400
+                        })
+                    })
+            });
+            it('case 4: unregistered username sent in request', () => {

```

Comment:
In the case of an unregistered user we should send back a 404, as the user is not found 🙂